Godeps/_workspace/src/k8s.io/kubernetes/pkg/api/v1/defaults.go

```diff
@@ -323,4 +324,42 @@ func defaultSecurityContextConstraints(scc *SecurityContextConstraints) {
 		scc.AllowEmptyDirVolumePlugin = new(bool)
 		*scc.AllowEmptyDirVolumePlugin = true
 	}
+
+	// defaults the volume slice of the SCC.
+	// In order to support old clients the boolean fields will always take precedence.
+	defaults := fsTypeToStringSet(scc.Volumes)
+
+	if scc.AllowHostDirVolumePlugin {
+		defaults.Insert(string(FSTypeHostPath))
+	} else {
+		defaults.Delete(string(FSTypeHostPath))
+	}
+
+	// should never be nil due to defaulting but check just in case
+	if scc.AllowEmptyDirVolumePlugin != nil && *scc.AllowEmptyDirVolumePlugin {
+		defaults.Insert(string(FSTypeEmptyDir))
+	} else {
+		defaults.Delete(string(FSTypeEmptyDir))
+	}
+
+	scc.Volumes = stringSetToFSType(defaults)
+}
+
+func stringSetToFSType(set sets.String) []FSType {
+	if set.Len() == 0 {
+		return nil
+	}
+	volumes := []FSType{}
+	for _, v := range set.List() {
+		volumes = append(volumes, FSType(v))
+	}
+	return volumes
+}
+
+func fsTypeToStringSet(volumes []FSType) sets.String {
```

Comment:
this becomes important when we remove the emptyDir bool... a nil volumes slice has to default to FSTypeEmptyDir for compatibility, but an empty slice can really mean no volume types are allowed